ui/components/src/ThresholdsEditor/ThresholdColorPicker.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { styled, IconButton, Popover } from '@mui/material';
+import CircleIcon from 'mdi-material-ui/Circle';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import { ColorPicker } from '../ColorPicker';
+import { ThresholdInputProps } from './ThresholdsEditor';
+
+export function ThresholdColorPicker({
+  color,
+  onColorChange,
+  label,
+}: Pick<ThresholdInputProps, 'color' | 'onColorChange' | 'label'>) {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
+  const isOpen = Boolean(anchorEl);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
```

Comment:
If felt like it, could do something like `openColorPicker` and `closeColorPicker` (and both internally call the `setAnchorEl`) (if I'm reading right, I think that's the main outcome for `setAnchorEl`)